reference/src/functions/macros.md

```diff
@@ -490,6 +489,71 @@ let result = {
 };
 ```
 
+This can often be helpful when using "loop"-like macros to break early. For example in the
+`vector::any` macro
+
+```move
+public macro fun any<$T>($v: &vector<$T>, $f: |&$T| -> bool): bool {
+    let v = $v;
+    'any: {
+        v.do_ref!(|e| if ($f(e)) return 'any true);
+        false
+    }
+}
+```
+
+The `return 'any true` exits from the "loop" early when the condition is met. Otherwise, the macro
+"returns" `false`.
+
+### Method Syntax
+
+When applicable, `macro` functions can be called using [method syntax](../method-syntax.md). When
+using method syntax, the evaluation of the arguments will change in that the first argument (the
+"receiver" of the method) will be evaluated outside of the macro expansion. This example is
+contrived, but will concisely demonstrate the behavior.
+
+```move
+public struct S() has copy, drop;
+public fun foo(): S { abort 0 }
+public macro fun maybe_s($s: S, $cond: bool): S {
+    if ($cond) $s
+    else S()
+}
+```
+
+Even though `foo()` will abort, it's return type can be used to start a method call.
+
+`$s` will not be evaluated if `$cond` is `false`. Son under a normal non-method call, an argument of
```

Comment:
Can't suggest and edit for some reason using GitHub machinery so here it is as text:
```
`$s` will not be evaluated if `$cond` is `false`, and under a normal non-method call, an argument of
```